### DISCUSSION TITLE
Album creation workflow

### DISCUSSION THREAD

**Discussion by LouisLouiseKay** [funmusicplace/mirlo#161] [Ideas]
Improvements propositions for the Album Creation workflow : 

- [x] Creating an album and adding tracks shouldn't be a two steps process. Clicking "add an album" should essentially take you to the equivalent of the page currently accessible after adding an album when you click "manage".
- [x] The album creation page shouldn't be a pop-up : the album creation process can be slow and require precision, it's not a transition process as you can spend a lot of time on that page. As such it'd be best to take full advantage of viewport to have a more balanced UI.
- [x] When you add a new track, while it's adding it, the "add track" button should display "uploading..." 
- [x] When you "publish an album", it should indicate "publishing..." or "in progress..." while it's publishing, before switching to "view live" once it's published.
- [x] When you add an image for album cover, there's no state indicator for if it's uploading or not.
- [x] "Types" drodown should have Single.
- [x] "Save Album" button should be "Save Draft" if an album is not published, and "Update" once an album is published.
- [x] "Publish" button should switch to "Make Private" or "Unpublish" once album is published.
- [x] There shouldn't be an "add track button", instead, "choose file" should be the "add track" button or "add audio file". Tracks addition should be saved by default after upload. Track information saved when "Save draft" is clicked.
- [x] The additional artist field names ("How the artist's name should appear publically | Is this an existing artist in Mirlo? Add their ID if so. | What role did the artist have?") should only appear once "add artist" is clicked on.
- [x] Album image is squeezed ![2023-10-15_155340](https://github.com/funmusicplace/mirlo/assets/77789607/1f627653-2c7f-43fe-8af4-ddc009e1591c) in the album management view. I wonder if the "manage artist" view shouldn't just be the "live" view, with "edit" buttons on all the editable elements (create tiers in the subscription section, edit album under each album etc.). Worth pondering pros and cons but it would be quicker to visualize change AND the live view is better.
- [x] Add a "credit field" (to appear as muted text) when editing an album for liner notes.
* [x] Adding a new track is a little unintuitive (have to delete previous title)
* [x] Not entirely clear what status dropdown means (preview v must own)?
* [x] Allow easy track re-ordering
* [x] Hovering over edit button on track when in album upload shows “delete”
* [x] Allow replacing a track audio when editing the track
* [x] “View live” button under Manage Artist Name takes me to mirlo.space

**Comment by simonv3** [comment]
Want to also link this ticket which is about bulk uploading of tracks https://github.com/funmusicplace/mirlo/issues/95

**Comment by simonv3** [comment]
I'm just going to do some grunt work on this but I think the final flow is still up for design

**Comment by simonv3** [comment]
@LouisLouiseKay 

>  There shouldn't be an "add track button", instead, "choose file" should be the "add track" button or "add audio file". Tracks addition should be saved by default after upload. Track information saved when "Save draft" is clicked.

Do you think this is still relevant with the current flow? I guess I'm asking - should people be able to edit their tracks before uploading or should we just upload right away after they select files? Mainly I'm asking because it takes a while for a track to be completely processed.

**Comment by simonv3** [comment]
>  Add a "credit field" (to appear as muted text) when editing an album for liner notes.

@LouisLouiseKay Is there a difference between a credit field and the way we do `trackArtists` right now? Is this an additional field? Is there a corresponding id3 tag that does this already?

**Reply by LouisLouiseKay** [reply]
Yes so the reason the credit field is important is because lots of people working on an album will not want to be featured artists or appear as co-author under the song. Mastering engineer, recording studios, mastering studio, who owns copyright of the songs, etc. There are lots of infos that you don't wanna display anywhere else than in liner notes. We COULD down the line how extremely detailed Metadata (the same way aggregators do it), with the downside that it makes for a much more cumbersome album creation process vs copy pasting your liner notes. But then we'd need to nail a huge number of fields and we would STILL need credits just for... The thanks section, which matter to a lot of artists.

**Reply by simonv3** [reply]
Yeah `trackArtist` are differentiated by `isCoAuthor` which determines if they're listed in the track listing or not. Any `trackArtist` without the `isCoAuthor` boolean wouldn't be listed there. 

But it sounds like you just want a free form text field.

**Comment by simonv3** [comment]
This has mostly been implemented. If something needs to be changed we should create an issue / new idea
